core/arch/aarch64/instr_create.h

```diff
@@ -369,8 +369,11 @@
  * they just need to know whether they need to preserve the app's flags, so maybe
  * we can just document that this may not write them.
  */
-#define XINST_CREATE_slr_s(dc, d, s) \
-  instr_create_1dst_2src((dc), OP_lsrv, (d), (d), (s))
+#define XINST_CREATE_slr_s(dc, d, rm_or_imm) \
+  opnd_is_reg(rm_or_imm) ? \
+    instr_create_1dst_2src((dc), OP_lsrv, (d), (d), (rm_or_imm)) : \
+    instr_create_1dst_3src((dc), OP_ubfm, (d), (d), (rm_or_imm), \
```

Comment:
Some musings:

1. The Travis failure comes from the `\param s` when the argument is now called `rm_or_imm`. It's not exactly obvious from the published build log, is it?

2. There are an awful lot of unnecessary parentheses here. There's no need to bracket a macro argument when it's sitting between two commas.

3. Line 376 has more than 90 characters. Something for the automatic style check?

4. Are we sure that `INSTR_CREATE_lsr` should accept immediate and register arguments? It's not clear to me what the rules are for INSTR_CREATE macros. I read somewhere (arm/instr_create.h) that there is one INSTR_CREATE_xxx for each OP_xxx and the macro supplies any implicit operands. It might be better to leave it at that rather than aim at accepting every "alias". If you did want to accept every alias, then `INSTR_CREATE_mov` might be a monster (MOV is officially an alias of ADD, DUP, INS, MOVN, MOVZ, ORR or UMOV).